integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -160,6 +163,31 @@ async def resync_folders(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                             yield folders_batch
 
 
+@ocean.on_resync(ObjectKind.FILE)
+async def resync_files(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    for service in get_cached_all_services():
+        gitlab_resource_config: GitLabFilesResourceConfig = typing.cast(
+            "GitLabFilesResourceConfig", event.resource_config
+        )
+        if not isinstance(gitlab_resource_config, GitLabFilesResourceConfig):
+            return
+
+        selector = gitlab_resource_config.selector
+
+        if not (selector.files and selector.files.path):
+            logger.warning("No path provided in the selector, skipping fetching files")
+            return
+
+        for service in get_cached_all_services():
+            async for projects_batch in service.get_all_projects():
+                for project in projects_batch:
+                    if service.should_process_project(project, selector.files.repos):
+                        async for files_batch in service.get_all_files_in_project(
+                            project, selector.files.path
+                        ):
+                            yield files_batch
+
+
```

Comment:
Trying to understand the need for another iteration here. why can't we use the services in obtained from the first iteration of `get_cached_all_services()` at line 168  ?

Reply:
That's correct. I missed that